### ISSUE TITLE
Topology Atom dict keys not updating

### ISSUE THREAD

**Issue by ehhartmann** [hits-mbm-dev/kimmdy#285]
for Topology.del_atom I get a self.atoms dict like this after `update_map_all = self.reindex_atomnrs()`

```
{'1': Atom(nr='1', type='CT', resnr='1', residue='ACE', atom='CH3', cgnr='1', charge='-0.3662', mass='12.01', typeB=None, chargeB=None, massB=None, bound_to_nrs=['2', '3', '4'], is_radical=True),
 '3': Atom(nr='2', type='HC', resnr='1', residue='ACE', atom='HH32', cgnr='3', charge='0.1123', mass='1.008', typeB=None, chargeB=None, massB=None, bound_to_nrs=['1'], is_radical=False),
 '4': Atom(nr='3', type='HC', resnr='1', residue='ACE', atom='HH33', cgnr='4', charge='0.1123', mass='1.008', typeB=None, chargeB=None, massB=None, bound_to_nrs=['1'], is_radical=False),
 '5': Atom(nr='4', type='C', resnr='1', residue='ACE', atom='C', cgnr='5', charge='0.5972', mass='12.01', typeB=None, chargeB=None, massB=None, bound_to_nrs=['1', '5', '6'], is_radical=False),
 '6': Atom(nr='5', type='O', resnr='1', residue='ACE', atom='O', cgnr='6', charge='-0.5679', mass='16', typeB=None, chargeB=None, massB=None, bound_to_nrs=['4'], is_radical=False), '7': Atom(nr='6', type='N', resnr='2', residue='TYR', atom='N', cgnr='7', charge='-0.4157', mass='14.01', typeB=None, chargeB=None, massB=None, bound_to_nrs=['4', '7', '8'], is_radical=False), '8': Atom(nr='7', type='H', resnr='2', residue='TYR', atom='H', cgnr='8', charge='0.2719', mass='1.008', typeB=None, chargeB=None, massB=None, bound_to_nrs=['6'], is_radical=False), '9': Atom(nr='8', type='CT', resnr='2', residue='TYR', atom='CA', cgnr='9', charge='-0.0014', mass='12.01', typeB=None, chargeB=None, massB=None, bound_to_nrs=['6', '9', '10', '25'], is_radical=False), '10': Atom(nr='9', type='H1', resnr='2', residue='TYR', atom='HA', cgnr='10', charge='0.0876', mass='1.008', typeB=None, chargeB=None, massB=None, bound_to_nrs=['8'], is_radical=False), '11': Atom(nr='10', type='CT', resnr='2', residue='TYR', atom='CB', cgnr='11', charge='-0.0152', mass='12.01', typeB=None, chargeB=None, massB=None, bound_to_nrs=['8', '11', '12', '13'], is_radical=False), '12': Atom(nr='11', type='HC', resnr='2', residue='TYR', atom='HB1', cgnr='12', charge='0.0295', mass='1.008', typeB=None, chargeB=None, massB=None, bound_to_nrs=['10'], is_radical=False), '13': Atom(nr='12', type='HC', resnr='2', residue='TYR', atom='HB2', cgnr='13', charge='0.0295', mass='1.008', typeB=None, chargeB=None, massB=None, bound_to_nrs=['10'], is_radical=False), '14': Atom(nr='13', type='CA', resnr='2', residue='TYR', atom='CG', cgnr='14', charge='-0.0011', mass='12.01', typeB=None, chargeB=None, massB=None, bound_to_nrs=['10', '14', '23'], is_radical=False), '15': Atom(nr='14', type='CA', resnr='2', residue='TYR', atom='CD1', cgnr='15', charge='-0.1906', mass='12.01', typeB=None, chargeB=None, massB=None, bound_to_nrs=['13', '15', '16'], is_radical=False), ...}
```

**Comment by jmbuhr** [comment]
Is this still an issue?

**Comment by ehhartmann** [comment]
Fixed in #331
